jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/ReservedThreadExecutor.java

```diff
@@ -217,225 +207,167 @@ public void execute(Runnable task) throws RejectedExecutionException
                     {
-                        long now = NanoTime.now();
-                        long lastEmpty = _lastEmptyNanoTime.get();
-                        if (size > 0 && _idleTimeNanos < NanoTime.elapsed(lastEmpty, now) && _lastEmptyNanoTime.compareAndSet(lastEmpty, now))
-                        {
-                            // it has been too long since we hit zero reserved threads, so are "busy" idle
-                            next = State.IDLE;
-                        }
-                        else
+                        if (!isRunning())
+                            return;
+
+                        // Shrink if we are already removed or there are other reserved threads;
+                        // there is a small chance multiple threads will shrink below minSize.
+                        if (getAvailable() > _minSize && _threads.remove(this, slot))
                         {
-                            // We will become a reserved thread if we can update the count below.
-                            next = State.RESERVED;
-                            size++;
+                            if (LOG.isDebugEnabled())
+                                LOG.debug("{} reservedThread shrank {}", ReservedThreadExecutor.this, this);
+                            return;
                         }
+                        task = waitForTask();
                     }
 
-                    // Update count for pending and size
-                    if (!_count.compareAndSet(count, pending, size))
-                        continue;
-
-                    if (LOG.isDebugEnabled())
-                        LOG.debug("{} was={} next={} size={}+{} capacity={}", this, _state, next, pending, size, _capacity);
-                    _state = next;
-                    if (next != State.RESERVED)
-                        break;
-
-                    // We are reserved whilst we are waiting for an offered _task.
-                    Runnable task = reservedWait();
-
-                    // Is the task the STOP poison pill?
-                    if (task == STOP)
-                        break;
-
-                    // Run the task
                     try
                     {
-                        _state = State.RUNNING;
+                        if (LOG.isDebugEnabled())
+                            LOG.debug("{} reservedThread run {} on {}", ReservedThreadExecutor.this, task, this);
                         task.run();
                     }
-                    catch (Throwable e)
+                    catch (Throwable t)
                     {
-                        LOG.warn("Unable to run task", e);
-                    }
-                    finally
-                    {
-                        // Clear any interrupted status.
-                        Thread.interrupted();
+                        if (LOG.isDebugEnabled())
+                            LOG.debug("{} task {} failure", ReservedThreadExecutor.this, task, t);
```

Comment:
Restore the `finally` block that clears the interrupted status.
It was the only log that was to be removed.

Reply:
Nah!   QTP already clears the interrupted status of every task, so we don't need to do it in every thread.    Only reason to do it here is to perhaps log that we were interrupted rather than exiting normally.   Calling interrupted and ignoring the result makes no sense.

Reply:
@gregw see #7548, you cannot remove that `finally` block.

Reply:
@gregw you forgot this.